test/libsolidity/syntaxTests/multiSource/free_function_redefinition_transitive.sol

```diff
@@ -0,0 +1,12 @@
+==== Source: s1.sol ====
+function f() pure returns (uint) { return 1337; }
+contract C {}
+==== Source: s2.sol ====
+import "s1.sol";
+contract D is C {}
+==== Source: s3.sol ====
+import "s2.sol";
+function f() pure returns (uint) { return 42; }
+contract E is D {}
+// ----
+// DeclarationError 1686: (s1.sol:0-49): Function with same name and parameter types defined twice.
```

Comment:
This error might be hard to read, because it points at the import statement. I think we currently do not have a better mechanism, but could be something for the future.

Reply:
Created an issue for this: https://github.com/ethereum/solidity/issues/9856